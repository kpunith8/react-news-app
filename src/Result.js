import React from "react";

// Higher order function, filters the items
/* const isSearched = searchTerm => item =>
   item.title.toLowerCase().includes(searchTerm.toLowerCase());
*/
// usage: items.filter(isSearched(searchPattern)).map(item => ())

const Result = ({ items, searchPattern }) => (
  <div className={"row"}>
    {items.map(item => (
      <div className={"row-item"} key={item.objectID}>{item.title}</div>
    ))}
  </div>
);

export default Result;
