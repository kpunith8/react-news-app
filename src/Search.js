import React from "react";

// Functional Stateless component or Presentational component
// Controlled component - since value of the input is controlled from parent component
const Search = ({ value, onChange, onSubmit, children }) => (
  <div>
    <form onSubmit={onSubmit}>
      <input type="text" value={value} onChange={onChange} />
      <button type="submit">{children}</button>
    </form>
  </div>
);

export default Search;
