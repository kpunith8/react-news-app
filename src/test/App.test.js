import React from "react";
import renderer from "react-test-renderer";
import Search from '../../src/Search';
import Result from '../../src/Result';
import Adapter from 'enzyme-adapter-react-16.3';
import Enzyme, { shallow } from 'enzyme';

Enzyme.configure({ adapter: new Adapter() });

const props = {
  items: [
    { title: '1', author: '1', num_comments: 1, points: 2, objectID: 'y' },
    { title: '2', author: '2', num_comments: 1, points: 2, objectID: 'z' },
  ],
};

describe("App", () => {

  test("Search has a valid snapshot", () => {
    const component = renderer.create(<Search>Search</Search>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Result has a valid snapshot', () => {
    const component = renderer.create(
      <Result {...props} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  // using enzyme
  it('Result shows two items in list', () => {
    const element = shallow(
      <Result {...props} />
    );
    expect(element.find('.row').length).toBe(1);
    expect(element.find('.row-item').length).toBe(2);
  });
});