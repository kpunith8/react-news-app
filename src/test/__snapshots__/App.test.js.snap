// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Result has a valid snapshot 1`] = `
<div
  className="row"
>
  <div
    className="row-item"
  >
    1
  </div>
  <div
    className="row-item"
  >
    2
  </div>
</div>
`;

exports[`App Search has a valid snapshot 1`] = `
<div>
  <form
    onSubmit={undefined}
  >
    <input
      onChange={undefined}
      type="text"
      value={undefined}
    />
    <button
      type="submit"
    >
      Search
    </button>
  </form>
</div>
`;
